pkg/icingadb/runtime_updates.go

```diff
@@ -346,8 +488,19 @@ func structifyStream(
 						return ctx.Err()
 					}
 				} else if runtimeType == "delete" {
+					if service, ok := entity.(*v1.Service); ok {
+						if service.HostId == nil {
+							err := scanHostIdForService(ctx, service, db)
+							if err != nil {
+								return err
+							}
+
+							entity = service
+						}
+					}
```

Comment:
Yes, it could be, but I just want to avoid an extra goroutine that receives the service, selects its host ID and forwards it to another one. Here on the other hand you don't need an extra step.

Reply:
And I would need to add a differentiation for service and host types. But this way I don't need special treatment for services.